### DISCUSSION TITLE
Update-All, if the user doing it is not owner of the files/folders, it will say "Failed to extract snapshots for the custom node"

### DISCUSSION THREAD

**Discussion by ita2444** [Comfy-Org/ComfyUI-Manager#1291] [General]
Hi,

I am setting up a centralized installation of Comfy for several people to collaborate with the same setup of nodes and models.
So the folders "custom_nodes" and "models" are symlinked (in linux) to a server.

When we hit "Update All", if the user doing it is not owner of the files/folders, it will say "Failed to extract snapshots for the custom node" for all custom nodes.
If the user is the owner, it will work.

Is that by design ?
Can I cange that is some way?

Thanks

**Comment by ltdrdata** [comment]
What you're describing seems to be a permission issue. This is not something that can be resolved within the custom node itself.
I plan to provide a feature in the future that allows changing the storage path for each user through the `--user-directory` option.

**Reply by ita2444** [reply]
Oh nice ! to have that option built-in !!

In regard to my problem, it is not a "permission" issue, because the permissions are 777 (fully writable).
I tested:
A) I am owner, it updates
B) not owner, it fails with "Failed to extract snapshots for the custom node"

Really weird.
If you say there is nothing in your code that checks for ownership of files, or anything in the likes... I will dig on my side.

Thanks a lot for your time and your contribution 👍

**Comment by ltdrdata** [comment]
The patch for moving dynamically generated files to the user directory has already been fully applied.
